File: HurbChallenge/HurbChallenge/Models/Filters.swift
//
//  Filters.swift
//  HurbChallenge
//
//  Created by Felipe Alves on 05/07/19.
//  Copyright © 2019 Bolzaniapps. All rights reserved.
//

import Foundation

struct Filters: Codable {
    let amenities: [FilterElement]
    let attributes: [FilterElement]
    let countries: [FilterElement]
    let cities: [FilterElement]
    let departureCities: [FilterElement]
    let district: [FilterElement]
    let neighborhood: [FilterElement]
    let duration: [FilterElement]
    let food: [FilterElement]
    let people: [FilterElement]
    let prices: [PriceElement]
    let priceInterval: PriceInterval
    let productType: [FilterElement]
    let regulation: [FilterElement]
    let rooms: [FilterElement]
    let stars: [FilterElement]
    let states: [FilterElement]
}

// MARK: -
struct FilterElement: Codable {
    let term: String
    let filter: String
    let count: Int
}

// MARK: -
struct PriceInterval: Codable {
    let min: Int
    let max: Int
    let filterPattern: String
}

// MARK: -
struct PriceElement: Codable {
    let min: Int
    let maxExclusive: Int
    let filter: String
    let count: Int
}

File: data/src/main/java/com/br/natanbrito/challenge/data/model/filters/Filters.kt
package com.br.natanbrito.challenge.data.model.filters

data class Filters(
    val amenities: List<Amenity>,
    val cities: List<City>,
    val countries: List<Country>,
    val priceInterval: PriceInterval,
    val prices: List<Price>,
    val productType: List<ProductType>,
    val rooms: List<Room>,
    val stars: List<Star>,
    val states: List<State>
)

data class Amenity(
    val count: Int,
    val filter: String,
    val term: String
)

data class City(
    val count: Int,
    val filter: String,
    val term: String
)

data class Country(
    val count: Int,
    val filter: String,
    val term: String
)

data class PriceInterval(
    val filterPattern: String,
    val max: Int,
    val min: Int
)

data class ProductType(
    val count: Int,
    val filter: String,
    val term: String
)

data class Room(
    val count: Int,
    val filter: String,
    val term: String
)

data class Star(
    val count: Int,
    val filter: String,
    val term: String
)

data class State(
    val count: Int,
    val filter: String,
    val term: String
)